master/buildbot/reporters/http.py

```diff
@@ -41,11 +41,28 @@ def checkConfig(self, *args, **kwargs):
                          (self.__class__,))
         if not isinstance(kwargs.get('builders'), (type(None), list)):
             config.error("builders must be a list or None")
+        # check that our callbacks are functions
+        if kwargs.get('startCB') and not hasattr(kwargs['startCB'], '__call__'):
+            config.error("start callback must be callable")
+        if kwargs.get('reviewCB') and not hasattr(kwargs['reviewCB'], '__call__'):
+            config.error("review callback must be callable")
+        if kwargs.get('summaryCB') and not hasattr(kwargs['summaryCB'], '__call__'):
+            config.error("summary callback must be callable")
 
     @defer.inlineCallbacks
-    def reconfigService(self, builders=None, **kwargs):
+    def reconfigService(self, builders=None,
+                        startCB=None, startArg=None,
+                        reviewCB=None, reviewArg=None,
+                        summaryCB=None, summaryArg=None,
+                        **kwargs):
         yield service.BuildbotService.reconfigService(self)
         self.builders = builders
+        self.startCB = startCB
+        self.startArg = startArg
+        self.reviewCB = reviewCB
```

Comment:
Ok. I'll remove the callbacks from the base class and move them into the specific reporter. I'll move the corresponding unit tests as well. 

Thanks for the feedback. The biggest piece that resonated with me was the piece about current use cases - design for what we know to avoid over engineering.